java/src/jmri/util/ThreadingUtil.java

```diff
@@ -224,6 +224,25 @@ static public boolean isGUIThread() {
         return SwingUtilities.isEventDispatchThread();
     }
 
+    /**
+     * Get the JMRI default thread group.
+     * This should be passed to as the first argument to the {@link Thread} constructor
+     * so we can track JMRI-created threads.
+     */
+    static public ThreadGroup getJmriThreadGroup() {
```

Comment:
It seems to me this method can create multiple ThreadGroups named "JMRI". Are we wanting to allow that, or are we wanting there to be just one ThreadGroup named "JMRI"?

Reply:
Once the "JMRI" group has been created on the first pass, this navigates the ThreadGroup tree to find the guaranteed "main" root, then down to find the "JMRI" group.